src/serac/physics/utilities/functional/domain_integral_cuda.cuh

```diff
@@ -0,0 +1,292 @@
+#pragma once
+
+#include "mfem.hpp"
+#include "mfem/linalg/dtensor.hpp"
+
+#include "serac/physics/utilities/functional/integral_utilities.hpp"
+#include "serac/physics/utilities/functional/domain_integral_shared.hpp"
+#include <cstring>
+
+namespace serac {
+
+namespace detail {
+
+/**
+ * @brief utility method to display last cuda error message
+ *
+ * @param[in] o The output stream to post success or CUDA error messages
+ * @param[in] success_string A string to print if there are no CUDA error messages
+ */
+inline void displayLastCUDAErrorMessage(std::ostream& o, const char* success_string = "")
+{
+  auto error = cudaGetLastError();
+  if (error != cudaError::cudaSuccess) {
+    o << "Last CUDA Error Message :" << cudaGetErrorString(error) << std::endl;
+  } else if (strlen(success_string) > 0) {
+    o << success_string << std::endl;
+  }
+}
+
+/**
+ * @brief Defines whether to loop by elements or quadrature points
+ */
+enum ThreadExecutionPolicy
+{
+  THREAD_PER_ELEMENT_QUADRATURE_POINT,
```

Comment:
Would `THREAD_PER_QUADRATURE_POINT` be appropriate? I think it's more descriptive.